src/Storages/VirtualColumnUtils.cpp

```diff
@@ -132,6 +148,19 @@ VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription
     add_virtual("_size", makeNullable(std::make_shared<DataTypeUInt64>()));
     add_virtual("_time", makeNullable(std::make_shared<DataTypeDateTime>()));
 
+    auto map = parseHivePartitioningKeysAndValues(path);
```

Comment:
Let's move checking `use_hive_partitioning` here, so we can avoid repeating this check in every engine